applications/serial_lte_modem/src/slm_at_commands.c

```diff
@@ -0,0 +1,580 @@
+{
+	int ret;
+	uint16_t size_limit;
+	uint16_t time_limit;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		ret = at_params_short_get(&at_param_list, 1, &size_limit);
+		if (ret) {
+			return ret;
+		}
+		ret = at_params_short_get(&at_param_list, 2, &time_limit);
+		if (ret) {
+			return ret;
+		}
+		if (datamode_time_limit < DATAMODE_TIME_LIMIT_MAX &&
+		    datamode_size_limit < DATAMODE_SIZE_LIMIT_MAX) {
+			datamode_size_limit = size_limit;
+			datamode_time_limit = time_limit;
+		} else {
+			return -EINVAL;
+		}
+		break;
+
+	case AT_CMD_TYPE_READ_COMMAND:
+		sprintf(rsp_buf, "#XDATACTRL: %d,%d\r\n",
+				datamode_size_limit, datamode_time_limit);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		ret = 0;
+		break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		sprintf(rsp_buf, "#XDATACTRL=<size_limit>,<time_limit>\r\n");
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		ret = 0;
+		break;
+
+	default:
+		break;
+	}
+
+	return ret;
+}
+
+/**@brief handle AT#XCLAC commands
+ *  AT#XCLAC
+ *  AT#XCLAC? not supported
+ *  AT#XCLAC=? not supported
+ */
+int handle_at_clac(enum at_cmd_type cmd_type);
+
+/* TCP proxy commands */
+int handle_at_tcp_filter(enum at_cmd_type cmd_type);
+int handle_at_tcp_server(enum at_cmd_type cmd_type);
+int handle_at_tcp_client(enum at_cmd_type cmd_type);
+int handle_at_tcp_send(enum at_cmd_type cmd_type);
+int handle_at_tcp_recv(enum at_cmd_type cmd_type);
+
+/* UDP proxy commands */
+int handle_at_udp_server(enum at_cmd_type cmd_type);
+int handle_at_udp_client(enum at_cmd_type cmd_type);
+int handle_at_udp_send(enum at_cmd_type cmd_type);
+
+/* Socket-type TCPIP commands */
+int handle_at_socket(enum at_cmd_type cmd_type);
+int handle_at_socketopt(enum at_cmd_type cmd_type);
+int handle_at_bind(enum at_cmd_type cmd_type);
+int handle_at_connect(enum at_cmd_type cmd_type);
+int handle_at_listen(enum at_cmd_type cmd_type);
+int handle_at_accept(enum at_cmd_type cmd_type);
+int handle_at_send(enum at_cmd_type cmd_type);
+int handle_at_recv(enum at_cmd_type cmd_type);
+int handle_at_sendto(enum at_cmd_type cmd_type);
+int handle_at_recvfrom(enum at_cmd_type cmd_type);
+int handle_at_getaddrinfo(enum at_cmd_type cmd_type);
+
+#if defined(CONFIG_SLM_NATIVE_TLS)
+int handle_at_xcmng(enum at_cmd_type cmd_type);
+#endif
+
+/* ICMP commands */
+int handle_at_icmp_ping(enum at_cmd_type cmd_type);
+
+/* FOTA commands */
+int handle_at_fota(enum at_cmd_type cmd_type);
+
+#if defined(CONFIG_SLM_GPS)
+int handle_at_gps(enum at_cmd_type cmd_type);
+#endif
+
+#if defined(CONFIG_SLM_FTPC)
+int handle_at_ftp(enum at_cmd_type cmd_type);
+#endif
+
+#if defined(CONFIG_SLM_MQTTC)
+int handle_at_mqtt_connect(enum at_cmd_type cmd_type);
+int handle_at_mqtt_publish(enum at_cmd_type cmd_type);
+int handle_at_mqtt_subscribe(enum at_cmd_type cmd_type);
+int handle_at_mqtt_unsubscribe(enum at_cmd_type cmd_type);
+#endif
+
+#if defined(CONFIG_SLM_HTTPC)
+int handle_AT_HTTPC_CONNECT(enum at_cmd_type cmd_type);
```

Comment:
Would be good to unify handlers style at some future cleanup (i. e. use lowercase).

Reply:
updated

Reply:
Fixed